api-tests/src/utils/jar-generator.ts

```diff
@@ -70,8 +69,23 @@ export const generateJarPayload = async (
   return payload;
 };
 
-export const encryptJarRequest = async (payload: JarRequest): Promise<string> =>
-  await new jose.CompactEncrypt(
+const getCoreBackEncKey = async () => {
+  const encKeys = await jwks();
+  const key = encKeys.keys.find((key) => key.use === "enc");
+
+  if (!key) {
+    throw Error("No public encryption key found for core-back");
+  }
+  return key;
+};
+
+export const encryptJarRequest = async (
+  payload: JarRequest,
+): Promise<string> => {
+  const encryptionKeyJwk = await getCoreBackEncKey();
+  const encKey = await jose.importJWK(encryptionKeyJwk, encAlg);
```

Comment:
Do you think it's still worth caching these at the module level, so we don't have to refetch this for every test case? (We wouldn't need to worry about expiry or anything, just store the result once)

The endpoint will be pretty quick, so not critical.

Reply:
good idea, thanks :) (I assume we don't need anything more complex like an expiry)

Reply:
Yeah, if the keys change while the tests are running then we must be very unlucky!